core/src/main/web/app/mainapp/mainapp.js

```diff
@@ -616,8 +641,18 @@
               disconnected = true;
             },
             onReconnected: function() {
-              bkSessionManager.reconnectEvaluators();
-              disconnected = false;
+              bkSessionManager.isSessionValid()
```

Comment:
Yes, the session becomes invalid. To be more precise: there is a short period in between the server crash and the launch of a new instance where the client is disconnected and during that time it polls for a connection. When the connection is finally made the client validates the session.